app/workers/virtual_agent_store_user_info_job.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class VirtualAgentStoreUserInfoJob
+  include Sidekiq::Worker
+
+  def perform(user_info, action_type, _meta)
+    kms = KmsEncrypted::Box.new(previous_versions: [{ key_id: Settings.lockbox.master_key }])
```

Comment:
I think that this needs to be the same object that you're encrypting with. I think if you add some tests to the VirtualAgentUserAccessRecord, you will see that the record doesn't get decrypted and save properly.

Reply:
Using the same encryption service instance throughout now, and added some tests to validate that